plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadataColumns.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import io.trino.spi.type.Type;
+import static io.trino.plugin.iceberg.ColumnIdentity.TypeCategory;
+import static io.trino.plugin.iceberg.ColumnIdentity.TypeCategory.PRIMITIVE;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+
+public enum IcebergMetadataColumns
+{
+    FILE_PATH(Integer.MAX_VALUE - 1, "$path", VARCHAR, PRIMITIVE);
```

Comment:
I don't know what the implication of statically reserving a field ID may be.
Did you investigate alternative options for avoiding to clash with actual field IDs?

Reply:
I think that number comes from here actually: https://iceberg.apache.org/spec/#reserved-field-ids

You should be able to reuse the iceberg `FILE_PATH.fieldId()`

Reply:
@alexjo2144 updated.